Adds new vox mercenary voidsuit

bigfunguss opened the pull request:
:cl: Funguss
rscadd: Vox mercenaries get some racial gear and can cycle mercenary voidsuits to fit them, though they spawn with a pre-cycled one, sprite credit to Raptor
tweak: The light mercenary rig has had its helmet sprite adjusted for vox. No more exposed beak.
maptweak: Adds a nitrogen canister to the Desperado
/:cl:

NullSnapshot commented on code/game/antagonist/outsider/mercenary.dm:
```diff
@@ -31,9 +31,12 @@ GLOBAL_DATUM_INIT(mercs, /datum/antagonist/mercenary, new)
-
-	var/decl/hierarchy/outfit/mercenary = outfit_by_type(/decl/hierarchy/outfit/mercenary)
-	mercenary.equip(player)
+	if(player.species && player.species.get_bodytype(player) == SPECIES_VOX)
+		var/decl/hierarchy/outfit/mercenary = outfit_by_type(/decl/hierarchy/outfit/mercenary/vox)
+		mercenary.equip(player)
+	else
+		var/decl/hierarchy/outfit/mercenary = outfit_by_type(/decl/hierarchy/outfit/mercenary)
+		mercenary.equip(player)
```
This entire part could probably be a little shorter if it's written as:
```suggestion
	var/decl/hierarchy/outfit/mercenary = outfit_by_type(/decl/hierarchy/outfit/mercenary)
	if(player.species && player.species.getbodytype(player) == SPECIES_VOX)
		mercenary = outfit_by_type(/decl/hierarchy/outfit/mercenary/vox)
	mercenary.equip(player)
```
9 times out of 10, the mercenary is going to not be vox, so it's probably safe to start with our default, and then perform the edge case check. Also gets rid of a bit of redundancy.

bigfunguss replied:
Done

NullSnapshot commented on code/modules/clothing/spacesuits/void/merc.dm:
```diff
@@ -47,4 +47,34 @@
\ No newline at end of file
+	tank = /obj/item/weapon/tank/oxygen
+
+/obj/item/clothing/head/helmet/space/vox/merc //based off the vox space helmets in order to display quills correctly
```
Instead of implementing the vox suit as it's own unique suit, I'd look into using the suit cycler instead. The merc ship is equipped with one, where the allowed species are defined in [code/game/machinery/suit_cycler_units.dm#L54](https://github.com/Baystation12/Baystation12/blob/dev/code/game/machinery/suit_cycler_units.dm#L54).

The way the suit cycler works is by calling refit_for_species. The base proc of which is [code/modules/clothing/clothing.dm#L116](https://github.com/Baystation12/Baystation12/blob/dev/code/modules/clothing/clothing.dm#L116).

Spookerton replied:
Instead of this duplicate definition, making sure to rename the icon states if required, do something like
```
/obj/item/clothing/suit/space/void/merc/prepared/vox/Initialize()
	. = ..()
	refit_for_species(SPECIES_VOX)
	helmet.refit_for_species(SPECIES_VOX)
```

bigfunguss committed:
Vox mercenary rework

The merc voidsuit can cycle to fit vox, and vox spawn with some racial gear, along with a pre-cycled suit. Also makes the helmet on the regular merc rig use the vox-specific sprite so that the beak no longer pops out.

bigfunguss commented:
The single conflict isn't showing on my end in Git, but I went ahead and downloaded the latest version of the repo and fixed the file manually.

SierraKomodo commented:
Github's still showing a file conflict with obj_head.dmi

bigfunguss commented:
> Github's still showing a file conflict with obj_head.dmi

Not entirely sure why it's doing that, I downloaded the latest version from the repo, put my changes on that, and force pushed since gitkraken isn't showing a conflict

bigfunguss commented:
updated and rebased my local repo, should be working fine now.

bigfunguss commented:
Rebasing removed all my work, going to redo it then open this back up.
